### PR TITLE
Introduce Spinner while restoring wallet

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-react/src/playground/App.tsx [hunk 1]
```diff
@@ -272,7 +273,7 @@ export const App = () => {
                 render={({history}) => <EnterPassword
                   hideModal={() => history.push('/')}
                   walletService={walletService}
-                  onConfirm={(password) => console.log({password})}/>}
+                  onConfirm={async (password) => {console.log({password});}}/>}
```
Comment (xitronix) [inline universal-login-react/src/playground/App.tsx:276]: why this is necessary?

File: universal-login-react/src/ui/Onboarding/EnterPassword.tsx [hunk 2]
```diff
@@ -9,19 +9,24 @@ import '../styles/base/enterPassword.sass';
...
 interface EnterPasswordProps {
   hideModal?: () => void;
   walletService: WalletService;
-  onConfirm: (password: string) => void;
+  onConfirm: (password: string) => Promise<void>;
 }
 
 export const EnterPassword = ({hideModal, walletService, onConfirm}: EnterPasswordProps) => {
   const [password, setPassword] = useInputField(isProperPassword, '');
+  const [busy, setBusy] = useState(false);
```
Comment (xitronix) [inline universal-login-react/src/ui/Onboarding/EnterPassword.tsx:22]: loading

File: universal-login-react/src/ui/Onboarding/EnterPassword.tsx [hunk 3]
```diff
@@ -9,19 +9,24 @@ import '../styles/base/enterPassword.sass';
...
+  const primaryButtonClassName = useClassFor('proceed-btn');
 
-  const onConfirmClick = () => {
+  const onConfirmClick = async () => {
     ensureNotFalsy(password, Error, 'Password missing');
-    onConfirm(password);
+    setBusy(true);
+    await onConfirm(password);
+    setBusy(false);
```
Comment (xitronix) [inline universal-login-react/src/ui/Onboarding/EnterPassword.tsx:29]: i think setBusy(false) is unnecessary

File: universal-login-react/src/playground/App.tsx (hunk 1 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/playground/App.tsx:276]: The interface of EnterPassword component has changed - now it requires a promise function

Comment (xitronix) [reply universal-login-react/src/playground/App.tsx:276]: Does it work? 
```ts
async (password) => console.log(password);
```

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 4]
```diff
@@ -18,6 +18,7 @@ export interface OnboardingProps {
   walletService: WalletService;
   domains: string[];
   onConnect?: () => void;
+  onLogin?: () => void;
```
Comment (xitronix) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx:21]: onSuccess

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 5]
```diff
@@ -102,7 +103,10 @@ export const Onboarding = ({emailFlow = false, ...props}: OnboardingProps) => {
             <EnterPassword
               hideModal={props.hideModal}
               walletService={props.walletService}
-              onConfirm={(password: string) => props.walletService.restoreWallet(password)}
+              onConfirm={async (password: string) => {
+                await props.walletService.restoreWallet(password);
+                props.onLogin && props.onLogin();
```
Comment (xitronix) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx:108]: props?.onLogin()

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 4 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx:21]: it is in Onboarding

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 5 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx:108]: props is not optional

Comment (xitronix) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx:109]: sorry : `props.onLogin.?()`

**Review by xitronix — APPROVED** [review]

**Pull request merged**